Added a copy button

jgrrdnd opened the pull request:
issue #3

jgrrdnd committed:
Added a copy to clipboard button

PrabhanshuAttri commented on index.html:
```diff
@@ -90,8 +90,10 @@ <h2 class="ui teal image header">
                             <div class="ui error message"></div>
-                            <div class="field hexData"></div>
-
+                            <div id="p1" class="field hexData"></div>
```
instead of using id 'p1', you can use class 'hexData' as this class is unique on this page.

PrabhanshuAttri commented on index.html:
```diff
@@ -116,6 +118,15 @@ <h2 class="ui teal image header">
         <script async src="https://www.googletagmanager.com/gtag/js?id=UA-67526856-7"></script>
+	<script>
+            function copyToClipboard(element) {
+            var $temp = $("<input>");
```
rename the variable from $temp to inputElement or inputEle.

PrabhanshuAttri replied:
You can create an onclick event in jquery for class `hexData` instead of a separate function.
Something like this.
```
$(".hexData").on("click", function() {
var ele = $("<input>");
$("body").append(ele);
ele.val($(element).text()).select();
document.execCommand("copy");
ele.remove();
}
```

In this case, there will be no need of editing line 93 and `onclick="copyToClipboard('#p1')"` can be removed from the button. Just make sure the code is intended properly inside curly braces. Test it and please let me know.

PrabhanshuAttri commented:
@jgrrdnd Please add screenshot and mention issue number. Thanks.

jgrrdnd committed:
Changed var and id to hexData

jgrrdnd committed:
add style to Copy button

jgrrdnd committed:
added class bttnColor

jgrrdnd committed:
Update index.html

jgrrdnd committed:
Update index.html

jgrrdnd committed:
Update index.html

jgrrdnd committed:
Update index.html

jgrrdnd committed:
Update index.html

jgrrdnd commented:
```
<div class="field hexData"></div>
<div class="button">
<button class="bttnColor">Copy</button>
</div>
```
```
$(".hexData").on("click", function() {
 var inputEle = $("<input>");
 $("body").append(inputEle);
 inputEle.val($(element).text()).select();
 document.execCommand("copy");
 inputEle.remove();
 });
```
I had changed the code to what I have here ^ and Im having an issue with the button copying the code and the encoding is not working. I added class bttnColor to style the Copy button.

PrabhanshuAttri commented:
@jgrrdnd change this line
```
$(".hexData").on("click", function() {
```
to
```
$(".copyButton").on("click", function() {
```

and change class of the button, `bttnColor`, to `copyButton`. The code I gave you copies the encoded URL if you click it (the encoded URL).

jgrrdnd commented:
Yes, but once I click on the encoded url, I receive an error in the console:
`index.html?url=:125 Uncaught ReferenceError: element is not defined`
    `at HTMLDivElement.<anonymous> (index.html?url=:125)`
    `at HTMLDivElement.dispatch (jquery-3.2.1.min.js:3)`
    `at HTMLDivElement.q.handle (jquery-3.2.1.min.js:3)`
`(anonymous) @ index.html?url=:125`
`dispatch @ jquery-3.2.1.min.js:3`
`q.handle @ jquery-3.2.1.min.js:3`




if I change `$(".hexData")` to `$(".copyButton")` the encoded url wont be transfer.
Why not change it how I first added it and just rename the function?

jgrrdnd committed:
changed click function

jgrrdnd committed:
removed div button

jgrrdnd committed:
removed .bttn

jgrrdnd committed:
added cursor-pointer

jgrrdnd committed:
Merge pull request #1 from jgrrdnd/gh-pages

function successfully works. removed copy button

PrabhanshuAttri commented:
@jgrrdnd Is it working fine now? Could you please undo changes in ` css/semantic.css`? Thanks

jgrrdnd commented:
yes it is. and I have removed  any css code i added

PrabhanshuAttri commented:
@jgrrdnd There are 2 files which show up in file changes. Could you please undo changes this file '`css/semantic.css`' ?
